package_managers.py

```diff
@@ -30,53 +30,68 @@ def update_package_manager_data(self, package_manager_urls):
-                num_php_downloads = self.php_downloads(url)
-            if 'https://packagist.org/packages/sendgrid/php-http-client' == url:
-                num_php_http_client_downloads = self.php_downloads(url)
-            if 'pypi' in url and 'sendgrid' in url:
-                num_python_downloads = self.python_downloads(url)
-            if 'pypi' in url and 'python_http_client' in url:
-                num_python_http_client_downloads = self.python_downloads(url)
-            if 'pypi' in url and 'open_source_library_data_collector' in url:
-                num_python_open_source_library_data_collector_downloads = self.python_downloads(url)
-            if 'rubygems' in url and 'sendgrid' in url:
-                num_ruby_downloads = self.ruby_downloads(url)
-            if 'rubygems' in url and 'http' in url:
-                num_ruby_http_client_downloads = self.ruby_downloads(url)
-
-        return self.update_db(num_total_csharp_downloads,
-                              num_nodejs_monthly_downloads,
-                              num_php_downloads,
-                              num_python_downloads,
-                              num_ruby_downloads,
-                              num_python_http_client_downloads,
-                              num_python_open_source_library_data_collector_downloads,
-                              num_ruby_http_client_downloads,
-                              num_csharp_http_client_downloads,
-                              num_php_http_client_downloads,
-                              num_node_http_client_downloads)
+            self.update_data_object(url, data)
+
+        return self.update_db(data["num_total_csharp_downloads"],
+                              data["num_nodejs_monthly_downloads"],
+                              data["num_php_downloads"],
+                              data["num_python_downloads"],
+                              data["num_ruby_downloads"],
+                              data["num_python_http_client_downloads"],
+                              data["num_python_open_source_library_data_collector_downloads"],
+                              data["num_ruby_http_client_downloads"],
+                              data["num_csharp_http_client_downloads"],
+                              data["num_php_http_client_downloads"],
+                              data["num_node_http_client_downloads"])
+
+    def update_data_object(self, url, data):
+        """Updates the data object with the counts of downloaded objects.
+
+        :param url: The URL of the package manager
+        :type url: string
+
+        :param data: The data object to be populated
+        :type data: dict
+
+        """
+        if 'https://www.nuget.org/packages/SendGrid' == url:
+                data["num_total_csharp_downloads"] = self.csharp_downloads(url)
+        if 'https://www.nuget.org/packages/SendGrid.CSharp.HTTP.Client' == url:
```

Comment:
Is there any reason why you wouldn't change some of these to `elif` statements? If `url == 'https://www.nuget.org/packages/SendGrid'`, it couldn't also be `'https://www.nuget.org/packages/SendGrid.CSharp.HTTP.Client'`. All of the checks using `==` would be prime candidates but some of the `if` statements which use `in` would have to be thought about more.

For example, there might be a `url` which satisfies both of these checks:

```python
        if 'pypi' in url and 'sendgrid' in url:
            data["num_python_downloads"] = self.python_downloads(url)
        if 'pypi' in url and 'python_http_client' in url:
            data["num_python_http_client_downloads"] = self.python_downloads(url)
```